src/sound/Sound.cpp

```diff
@@ -588,125 +195,107 @@ namespace Sound {
 		std::map<std::string, Sample> m_loadedSounds;
 	};
 
-	std::vector<std::string> s_audioDeviceNames = {};
+	std::vector<std::string_view> GetAvailableBackends()
+	{
+		std::vector<std::string_view> backends{ SDLBackendName };
+#ifdef PI_BUILD_WITH_OPENAL
+		backends.push_back(OpenALBackendName);
+#endif
+		return backends;
+	}
 
-	bool Init(bool automaticallyOpenDevice)
+	std::string_view GetBackend()
 	{
-		PROFILE_SCOPED()
-		if (m_audioDevice) {
-			DestroyAllEvents();
-			return true;
+		if (dynamic_cast<SdlAudioBackend *>(m_backend) != nullptr) {
+			return SDLBackendName;
+		} else if (dynamic_cast<AlAudioBackend *>(m_backend) != nullptr) {
+			return OpenALBackendName;
 		}
+		return "Unknown Sound Backend";
+	}
 
-		if (SDL_Init(SDL_INIT_AUDIO) == -1) {
-			Output("Could not initialize SDL Audio: %s.\n", SDL_GetError());
-			return false;
+	bool Init(std::string_view backend)
+	{
+		PROFILE_SCOPED()
+		if (m_backend != nullptr) {
+			if (backend == GetBackend()) {
+				DestroyAllEvents();
+				return true;
+			} else {
+				Uninit();
+			}
 		}
 
-		// load all the wretched effects
-		Pi::GetApp()->GetAsyncStartupQueue()->Order(new LoadSoundJob("sounds", false));
-
-		//I'd rather do this in MusicPlayer and store in a different map too, this will do for now
-		Pi::GetApp()->GetAsyncStartupQueue()->Order(new LoadSoundJob("music", true));
-
-		UpdateAudioDevices();
-
-		// If we're going to manually pick a device later, don't open a default one now.
-		if (!automaticallyOpenDevice) {
-			return true;
+		if (backend.empty()) {
+#ifdef PI_BUILD_WITH_OPENAL
+			backend = OpenALBackendName;
+#else
+			backend = SDLBackendName;
+#endif
 		}
 
-		SDL_AudioSpec wanted;
-		wanted.freq = FREQ;
-		wanted.channels = 2;
-		wanted.format = AUDIO_S16;
-		wanted.samples = BUF_SIZE;
-		wanted.callback = fill_audio;
-		wanted.userdata = 0;
-
-		// Automatically pick the best device.
-		// TODO: allow the user to select which device they'd like to use.
-		m_audioDevice = SDL_OpenAudioDevice(nullptr, 0, &wanted, nullptr, 0);
-		if (!m_audioDevice) {
-			Output("Could not open audio device: %s\n", SDL_GetError());
+		try {
+			if (backend == SDLBackendName) {
+				m_backend = new SdlAudioBackend();
+			}
+#ifdef PI_BUILD_WITH_OPENAL
+			else if (backend == OpenALBackendName) {
+				m_backend = new AlAudioBackend();
+			}
+#endif
```

Comment:
If backends are `enum` instead of strings we could do a `switch` here..

Even better if they would be instance/proxy pointers, then you can just directly invoke the function and shold be able to keep the `Sound` class generic even if we were to add new backends in the future.

Reply:
This method is not called often, so I don't think string comparison is hurtful here.

About instance pointers, please read  https://github.com/pioneerspacesim/pioneer/pull/6188#discussion_r2274362482

About proxy pointers: that might work, but does it really worth the effort and the additional complexity?